### DISCUSSION TITLE
Grouped Minor Updates and Split Major

### DISCUSSION THREAD

**Discussion by CCMackenzie** [renovatebot/renovate#19654] [Request Help] [answered]
Hi,

I am trying to implement a solution using Renovate where the bot will group all minor updates according to their package manager, while keeping any major updates separate (i.e. one update per PR). 

From what I understand from the docs the options separateMajorMinor and separateMultipleMajor are set to true by default, I have explicitly setting these as true but still have grouped suggested PRs in the onboarding PRs for my repos.

Am I missing a setting that would make this possible? Any advice would be greatly appreciated.

Thanks!

**Comment by rarkins** [comment, marked as answer]
What's your config? Sounds like you need to use matchUpdateTypes as part of a packageRules config

**Reply by CCMackenzie** [reply]
I am using self-hosting on GitLab.

In my package rules I have:
<img width="322" alt="Screenshot 2023-01-04 at 10 50 03" src="https://user-images.githubusercontent.com/105081731/210539963-bd673f77-f348-4a24-842c-c828fe2b0867.png">
which is correctly grouping my updates by manager but also grouping all major updates into one PR. I also have the general config extending "config:base".

**Reply by rarkins** [reply]
Add matchUpdateTypes=minor,patch to that rule

**Reply by CCMackenzie** [reply]
That has sorted it. Thank you!

**Reply by Kurt-von-Laven** [reply]
Are major versions excluded from groups by default now? Trying to work out why mypy isn't grouped (c.f., ScribeMD/slack-templates#201) when it seems like it should be based on [our config](https://github.com/ScribeMD/.github/blob/1ae39d3fcd8da8ed043df2f1a1c38f2452ee18ee/default.json#L48-L58).

**Reply by viceice** [reply]
yes, checkout separate major minor option

**Comment by lapwat** [comment]
There exists a group preset that can handle that for you.

```txt
{
  extends: [
    'group:allNonMajor'
  ]
}
```

[https://docs.renovatebot.com/presets-group/#groupallnonmajor](https://docs.renovatebot.com/presets-group/#groupallnonmajor)
